MoreLinq/AggregateRight.cs

```diff
@@ -0,0 +1,117 @@
+﻿#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2016 Leandro F. Vieira (leandromoh). All rights reserved.
+// 
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// 
+//     http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+
+    static partial class MoreEnumerable
+    {
+        /// <summary>
+        /// Applies an right-associative accumulator function over a sequence.
+        /// </summary>
+        /// <typeparam name="TSource">The type of the elements of source.</typeparam>
+        /// <param name="source">Source sequence.</param>
+        /// <param name="func">An right-associative accumulator function to be invoked on each element.</param>
+        /// <returns>The final accumulator value.</returns>
+        /// <example>
+        /// <code>
+        /// var numbers = new int[] { 8, 12, 24, 4, 2 };
+        /// int result = numbers.AggregateRight((a, b) => a / b);
+        /// </code>
+        /// The <c>result</c> variable will contain <c>8</c>.
+        /// </example>
```

Comment:
A `<remarks>` section could be added noting this method iterates over the source sequence immediately (for all overloads). That same section could have a note detailing the relation with `Aggregate`.